backend/app/controllers/spree/admin/taxons_controller.rb

```diff
@@ -28,37 +28,41 @@ def edit
       end
 
       def update
-        parent_id = params[:taxon][:parent_id]
-        set_position
-        set_parent(parent_id)
+        successful = @taxon.transaction do
 
-        @taxon.save!
+          parent_id = params[:taxon][:parent_id]
+          set_position
+          set_parent(parent_id)
 
-        # regenerate permalink
-        regenerate_permalink if parent_id
+          @taxon.save!
 
-        set_permalink_params
+          # regenerate permalink
+          regenerate_permalink if parent_id
 
-        #check if we need to rename child taxons if parent name or permalink changes
-        @update_children = true if params[:taxon][:name] != @taxon.name || params[:taxon][:permalink] != @taxon.permalink
+          set_permalink_params
 
-        if @taxon.update_attributes(taxon_params)
-          flash[:success] = flash_message_for(@taxon, :successfully_updated)
-        end
+          #check if we need to rename child taxons if parent name or permalink changes
+          @update_children = true if params[:taxon][:name] != @taxon.name || params[:taxon][:permalink] != @taxon.permalink
```

Comment:
@charliestarkpl I believe you don't have to implicitly set `true` here depending on this `if-condition` result. This expression will returns a boolean value, so you can make advantage of it, and simply write:
`@update_children = (params[:taxon][:name] != @taxon.name || params[:taxon][:permalink] != @taxon.permalink)`
and you will also get rid of the hound warning :)